src/cpu/ppc64/ppc64_gemm_s8x8s32.cpp

```diff
@@ -0,0 +1,185 @@
+            const signed char *bo_flip = (const signed char *)bo;
+            for (int i = 0; i < b_size; ++i)
+                Bshort[i] = ((short)(Bflip[i])) - (short)*bo_flip;
+        } else {
+            for (int i = 0; i < b_size; ++i)
+                Bshort[i] = ((short)B[i]) - (short)*bo;
+        }
+        APraw = (short *)malloc((m_cap * k_cap + 15) * sizeof(short), 4096);
+        BPraw = (short *)malloc((k_cap * n_cap + 15) * sizeof(short), 4096);
+        if (utils::any_null(APraw, BPraw)) {
+            free(APraw);
+            free(BPraw);
+            return dnnl_out_of_memory;
+        }
+        AP = (short *)((((unsigned long)APraw) + 15) & (~15));
+        BP = (short *)((((unsigned long)BPraw) + 15) & (~15));
+        if (ATflag)
+            pack_N16_16bit(k, m, Ashort, lda, AP);
+        else
+            pack_T16_16bit(k, m, Ashort, lda, AP);
+        if (BTflag)
+            pack_T8_16bit(k, n, Bshort, ldb, BP);
+        else
+            pack_N8_16bit(k, n, Bshort, ldb, BP);
+        gemm_kernel_16bit(m, n, k, (float)alpha, AP, BP, C, beta, ldc);
+        free(Ashort);
+        free(Bshort);
+        free(APraw);
+        free(BPraw);
+    } else {
+        signed char *AP, *APraw;
+        unsigned char *BP, *BPraw;
+        APraw = (signed char *)malloc(
+                (m_cap * k_cap + 3) * sizeof(unsigned char), 4096);
+        BPraw = (unsigned char *)malloc(
+                (k_cap * n_cap + 3) * sizeof(signed char), 4096);
+        if (utils::any_null(APraw, BPraw)) {
+            free(APraw);
+            free(BPraw);
+            return dnnl_out_of_memory;
+        }
+        AP = (signed char *)((((unsigned long)APraw) + 3) & (~3));
+        BP = (unsigned char *)((((unsigned long)BPraw) + 3) & (~3));
+        if (ATflag)
+            pack_N16_8bit(k, m, A, lda, AP);
+        else
+            pack_T16_8bit(k, m, A, lda, AP);
+        if (flipB_flag) {
+            int b_size = ldb * (BTflag ? k - 1 : n - 1) + (BTflag ? n : k);
+            uint8_t *Bflip = (uint8_t *)malloc(b_size * sizeof(uint8_t), 4096);
+            if (utils::any_null(Bflip)) {
+                free(Bflip);
+                return dnnl_out_of_memory;
+            }
+            for (int i = 0; i < b_size; ++i)
+                Bflip[i] = B[i] ^ 0x80;
+            if (BTflag)
+                pack_T8_8bit(k, n, Bflip, ldb, BP);
+            else
+                pack_N8_8bit(k, n, Bflip, ldb, BP);
+            free(Bflip);
+        } else {
+            if (BTflag)
+                pack_T8_8bit(k, n, B, ldb, BP);
+            else
+                pack_N8_8bit(k, n, B, ldb, BP);
+        }
+        gemm_kernel_8bit(m, n, k, (float)alpha, AP, BP, C, beta, ldc);
+        if (flipB_flag) {
+            int *comparray = (int *)malloc(m * sizeof(int), 4096);
+            if (utils::any_null(comparray)) {
+                free(comparray);
+                return dnnl_out_of_memory;
```

Comment:
And here.

Reply:
My apologies.  Thank you for your patience in explaining this to me.
I have dropped code that I hope you will agree fixes this memory leak issue.